src/abi/argument-encoding.md

```diff
@@ -243,3 +252,66 @@ Calling `bar` with `MySumType::Z` yields:
 0x
 0000000000000002 // The discriminant of the chosen enum, in this case `2`.
 ```
+
+## Vectors
+
+ABI calls containing vectors are encoded in the following way:
+
+- First, figure out the the length `l` of the vector. Its length will also be its capacity.
+- Encode the content of the vector according to the spec of its type, e.g. for a `Vec<bool>`,
+  encode each `bool` element according to the `bool` specs. This gives out data that is stored
+  on the heap, and we encode only the pointer to this data
+
+```rust
+abi MyContract {
+  fn foo(a: Vec<u32>);
+} {
+  fn foo(a: Vec<u32>) {};
+}
+```
+
+Calling `foo` with `vec![1u32, 2u32, 3u32, 4u32]`:
+
+- `length` is 4, `capacity` is 4
+- `data`: [0x0000000000000001, 0x0000000000000002, 0x0000000000000003, 0x0000000000000004], stored at pointer address `0x000000000000beef`
+
+> Note: to understand how the `u32` are encoded, see the section about encoding integers.
+
+```plaintext
+0x
+000000000000beef // pointer address
+0000000000000004 // length = 4
+0000000000000004 // capacity = 4
+```
+
+At the pointer address, then the vector's content are encoded as such:
+
+```plaintext
+0x
+0000000000000001 // 1u32
+0000000000000002 // 2u32
+0000000000000003 // 3u32
+0000000000000004 // 4u32
+```
+
+## Tuples
+
+ABI calls containing tuples are encoded as such:
+If `X` is a tuple with the type signature `(T_1, T_2, ..., T_n)`, with `T_1,...,T_n` being any type except for vector, then `enc(X)` is encoded as the concatenation of `enc(T_1)`, `enc(T_2)`,`enc (T_3)`, ..., `enc(T_n)`.
+
+```rust
+abi MyContract {
+  fn foo(a: (u64, str[3], bool));
+} {
+  fn foo(a: (u64, str[3], bool)) {};
+}
+```
+
+Calling `foo` with `(1u64, "hel", true)` :
```

Comment:
maybe we can use another `str` here like `hello` or `fuel`. I do not know why `hel` was chosen :smile: